File: cpp/code program/motor.cpp
#pragma once
#include <iostream>
#include <string>
#include "kendaraan.cpp"

using namespace std;

class Motorcycle : public Kendaraan
{
private:
    string type;
    bool hasWindshield;

public:
    Motorcycle()
    {

    }

    Motorcycle(string b, string m, int y, double p, string t, bool hw): Kendaraan(b, m, y, p)
    {
        this->type = t;
        this->hasWindshield = hw;
    }

    void setType(string type)
    {
        this->type = type;
    }


    void setHasWindshield(bool hasWindshield)
    {
        this->hasWindshield = hasWindshield;
    }

    string getType()
    {
        return this->type;
    }


    bool getHasWindshield()
    {
        return this->hasWindshield;
    }

    ~Motorcycle()
    {

    }
};

File: python/code program/motor.py
from kendaraan import Kendaraan

class Motorcycle(Kendaraan):
    def __init__(self, brand="", model="", year=0, price=0.0, type_="", has_windshield=False):
        super().__init__(brand, model, year, price)
        self.type = type_
        self.has_windshield = has_windshield

    def set_type(self, type_):
        self.type = type_

    def set_has_windshield(self, has_windshield):
        self.has_windshield = has_windshield

    def get_type(self):
        return self.type

    def get_has_windshield(self):
        return self.has_windshield
